controllers/component_build_controller.go

```diff
@@ -487,6 +578,8 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 	log.Info(fmt.Sprintf("updated component after build request: %s", requestedAction), l.Action, l.ActionUpdate)
+	// remove component from metrics map
+	delete(componentTimesForMetrics, componentIdForMetrics)
```

Comment:
Is there a specific reason to delete the component from the global map _after_ doing the `Update` above, where we remove the request annotation? If we made it all the way here in the code, then the simple build / the PaC provision / etc. already succeeded no?

Reply:
if component isn't updated with removed request annotation 

and if update fails it will trigger another reconcile which will process request again

Reply:
Ah, that makes sense.

I guess the question is what exactly we want the SLIs to represent. The way they were worded originally, the time until build submission / PR submission would be what we want to measure (and also what the user probably cares about).

> Time taken from simple build request till the build pipeline is submitted.

> Time taken from PaC provision request till Component is provisioned for PaC builds. The last step (trigger) is that PR is created in GitHub.

But counting the time until the annotation is removed makes sense to me as well, since the controller's work technically isn't done until then. @mmorhun any thoughts?

Reply:
keep in mind that time difference between "action done" and "removing annotation" is minimal

also IF we would count success before removing annotation we would could be counting (build which failed on removal of annotation) multiple times

Reply:
> also IF we would count success before removing annotation we would could be counting (build which failed on removal of annotation) multiple times

Good point :+1: